website/components/navigation/Navigation.tsx

```diff
@@ -19,57 +23,73 @@ const logoStyles = css`
 `;
 
 function Navigation() {
+  const viewport = useViewportSize();
+  const isMobile = viewport && viewport.width < 768;
+  const Group = isMobile ? MobileNavigationGroup : SideNavGroup;
+  const Item = isMobile ? MobileNavigationItem : SideNavItem;
+
+  const groupProps =
+    !isMobile &&
+    ({
+      collapsible: true,
+      initialCollapsed: false,
+    } as const);
+
+  const content = (
+    <>
+      <Group header="Core Guidelines" {...groupProps}>
+        <Item>Logos</Item>
+        <Item>User Personas</Item>
+        <Item>Tone</Item>
+        <Item>Colors</Item>
+        <Item>Illustration</Item>
+        <Item>Typography</Item>
+      </Group>
+      <Group header="Components" {...groupProps}>
+        <Item>Badges</Item>
+        <Item>Banners</Item>
+        <Item>Box</Item>
+        <Item>Button</Item>
+        <Item>Callout</Item>
+        <Item>Code</Item>
+        <Item>Confirmation Modal</Item>
+        <Item>Icons</Item>
+        <Item>Icon Button</Item>
+        <Item>Inline Definition</Item>
+        <Item>Logo</Item>
+        <Item>Marketing Modal</Item>
+        <Item>Menu</Item>
+        <Item>Modal</Item>
+        <Item>Mongo Nav</Item>
+        <Item>Palette</Item>
+        <Item>Pipeline</Item>
+        <Item>Popover</Item>
+        <Item>Portal</Item>
+        <Item>Radio Box Group</Item>
+        <Item>Radio Group</Item>
+        <Item>Side Nav</Item>
+        <Item>Stepper</Item>
+        <Item>Syntax</Item>
+        <Item>Table</Item>
+        <Item>Tabs</Item>
+        <Item>Text Input</Item>
+        <Item>Toast</Item>
+        <Item>Toggle</Item>
+        <Item>Tokens</Item>
+        <Item>Tooltip</Item>
+        <Item>Typography</Item>
+      </Group>
+    </>
+  );
+
+  if (isMobile) {
```

Comment:
this definition has a lot of branching based on `isMobile` which makes it harder to reason about. I would explore what it would look like if there was only one branching point.

Reply:
The main thing I wanted to do here was share the the `content` between the Mobile and desktop Navigation components.

Reply:
Is it easier to make sense of if I move all of the conditional logic to a component, so it only reads as such:

```
function Navigation() {
  const viewport = useViewportSize();
  const isMobile = !!viewport && viewport.width < 768;

  if (isMobile) {
    return (
      <MobileNavigation>
        <Content isMobile />
      </MobileNavigation>
    );
  }

  return (
    <nav className={navWidth}>
      <MDBDesignLogo className={logoStyles} />
      <SideNav>
        <Content />
      </SideNav>
    </nav>
  );
}
```

Reply:
yeah, that could be an improvement

Reply:
oh I see you already made the change. 👍 LGTM

Reply:
👍